app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala

```diff
@@ -263,10 +267,42 @@ class CatalogueController @Inject() (
       .verify(Retrieval.hasPredicate(EnableBranchProtection.permission(repoName)))
       .map(r => EnableBranchProtection.HasAuthorisation(r.getOrElse(false)))
 
+  def changePrototypePassword(repoName: String) =
+    auth
+      .authorizedAction(
+        continueUrl = routes.CatalogueController.repository(repoName),
+        predicate = ChangePrototypePassword.permission(repoName)
+      ).async { implicit request =>
+
+      (for {
+        hasBranchProtectionAuth <- EitherT.liftF[Future, Result, EnableBranchProtection.HasAuthorisation](hasEnableBranchProtectionAuthorisation(repoName))
```

Comment:
we could move the `retrieval` into `authorizedAction` above, and save a trip to internal-auth